src/Microsoft.ML.PipelineInference/AutoInference.cs

```diff
@@ -62,16 +62,16 @@ public class EntryPointGraphDef
 
             /// <summary>
             /// Get the name of the variable asssigned to the Data or Training Data input, based on what is the first node of the subgraph.
-            /// A better way to do this would be with a ICanBeSubGraphFirstNode common interface between ITransformInput and ITrainerInputs 
-            /// and a custom deserializer. 
+            /// A better way to do this would be with a ICanBeSubGraphFirstNode common interface between ITransformInput and ITrainerInputs
+            /// and a custom deserializer.
             /// </summary>
             public string GetSubgraphFirstNodeDataVarName(IExceptionContext ectx)
             {
                 var nodes = Graph.GetNodes();
 
-                ectx.CheckValue(nodes, nameof(nodes), "Empty Subgraph");
-                ectx.CheckValue(nodes[0], nameof(nodes), "Empty Subgraph");
-                ectx.CheckValue(nodes[0][FieldNames.Inputs], "Inputs", "Empty subgraph node inputs.");
+                ectx.Check(nodes != null, "Empty Subgraph");
+                ectx.Check(nodes[0] != null, "Empty Subgraph");
```

Comment:
Having the same message as the above line makes it hard to diagnose which one of these checks actually failed.  Was `nodes` null?  or Was `nodes[0]` null?

Reply:
Fair point